pennylane/circuit_drawer/mpldrawer.py

```diff
@@ -0,0 +1,615 @@
+            **default_text_kwargs
+        )
+
+    def ctrl(self, layer, wire_ctrl, wire_target=None, control_values=None, color=None):
+        """Add an arbitrary number of control wires
+
+        Args:
+            layer (Int): the layer to draw the object in
+            wire_ctrl (Union[Int, Iterable[Int]]): set of wires to control on
+
+        Keyword Args:
+            wire_target=None (Union[Int, Iterable[Int]]): target wires. Used to determine min
+                and max wires for the vertical line
+            control_values=None (Iterable[Bool]): for each control wire, denotes whether to control
+                on ``False=0`` or ``True=1``.
+            kwargs=None (dict): mpl line keywords
+
+        **Example**
+
+        .. code-block:: python
+
+            drawer = MPLDrawer(n_wires=2, n_layers=3)
+
+            drawer.ctrl(layer=0, wire_ctrl=0, wire_target=1)
+            drawer.ctrl(layer=1, wire_ctrl=(0,1), control_values=[0,1])
+            drawer.ctrl(layer=2, wire_ctrl=(0,1), color="indigo")
+
+        .. figure:: ../../_static/drawer/ctrl.png
+            :align: center
+            :width: 60%
+            :target: javascript:void(0);
+
+        """
+        wire_ctrl = _to_tuple(wire_ctrl)
+        wire_target = _to_tuple(wire_target)
+
+        wires_all = wire_ctrl + wire_target
+        min_wire = min(wires_all)
+        max_wire = max(wires_all)
+
+        line = plt.Line2D((layer, layer), (min_wire, max_wire), zorder=2, color=color)
+        self.ax.add_line(line)
+
+        if control_values is None:
+            for wire in wire_ctrl:
+                self._ctrl_circ(layer, wire, zorder=3, color=color)
+        else:
+            if len(control_values) != len(wire_ctrl):
+                raise ValueError('`control_values` must be the same length as `wire_ctrl`')
+            for wire, control_on in zip(wire_ctrl, control_values):
+                if control_on:
+                    self._ctrlo_circ(layer, wire, zorder=3, color=color)
+                else:
+                    self._ctrl_circ(layer, wire, zorder=3, color=color)
+
+    def _ctrl_circ(self, layer, wire, zorder=3, color=None):
+        """Draw a solid circle that indicates control on one"""
+    
+        circ_ctrl= plt.Circle((layer, wire), radius=self._ctrl_rad, zorder=zorder, color=color)
+        self.ax.add_patch(circ_ctrl)
+
+    def _ctrlo_circ(self, layer, wire, zorder=3, color=None):
+        """Draw an open circle that indicates control on zero."""
+        kwargs = {
+            'edgecolor': plt.rcParams['lines.color'],
+            'facecolor': plt.rcParams['axes.facecolor'],
+            'linewidth': plt.rcParams['lines.linewidth']
+        }
+        if color is not None:
+            kwargs['edgecolor'] = color
+
+        circ_ctrlo = plt.Circle((layer, wire), radius=(1.5*self._ctrl_rad), zorder=zorder, **kwargs)
+
+        self.ax.add_patch(circ_ctrlo)
+
+    def CNOT(self, layer, wires, color=None):
```

Comment:
SWAP supports a great deal more customization as it supports any `Line2D` attribute, like linestyle, linewidth, or color.

I tried to support arbitrary keyword dictionaries on the control-type methods, but I couldn't find a good solution with the different types of keywords for the lines, filled circles, and background-filled circles.

Maybe I could accept a `lines_kwargs` dictionary and a `circle_kwargs` dictionary?

Reply:
I agree with @glassnotes on this.  The customization options should at least be consistent in all functions, so, if in this case `color` is the only option, it could be passed as `{'color': 'black}`. Do you mean that there would be other customization options that can be passed to either the line, circle or cross that warrants several dictionaries?